runtime/common/RuntimeMLIR.cpp

```diff
@@ -398,26 +419,53 @@ qirProfileTranslationFunction(const char *qirProfile, Operation *op,
                             "dynamic_qubit_management", falseValue);
   llvmModule->addModuleFlag(llvm::Module::ModFlagBehavior::Error,
                             "dynamic_result_management", falseValue);
+  if (isAdaptiveProfile) {
+    auto trueValue =
+        llvm::ConstantInt::getTrue(llvm::Type::getInt1Ty(*llvmContext));
+    llvmModule->addModuleFlag(llvm::Module::ModFlagBehavior::Error,
+                              "qubit_resetting", trueValue);
+    llvmModule->addModuleFlag(llvm::Module::ModFlagBehavior::Error,
+                              "classical_ints", falseValue);
+    llvmModule->addModuleFlag(llvm::Module::ModFlagBehavior::Error,
+                              "classical_floats", falseValue);
+    llvmModule->addModuleFlag(llvm::Module::ModFlagBehavior::Error,
+                              "classical_fixed_points", falseValue);
+    llvmModule->addModuleFlag(llvm::Module::ModFlagBehavior::Error,
+                              "user_functions", falseValue);
+    llvmModule->addModuleFlag(llvm::Module::ModFlagBehavior::Error,
+                              "dynamic_float_args", falseValue);
+    llvmModule->addModuleFlag(llvm::Module::ModFlagBehavior::Error,
+                              "extern_functions", falseValue);
+    llvmModule->addModuleFlag(llvm::Module::ModFlagBehavior::Error,
+                              "backwards_branching", falseValue);
+  }
 
   // Note: optimizeLLVM is the one that is setting nonnull attributes on
   // the @__quantum__rt__result_record_output calls.
   cudaq::optimizeLLVM(llvmModule.get());
   if (!cudaq::setupTargetTriple(llvmModule.get()))
     throw std::runtime_error("Failed to setup the llvm module target triple.");
 
+  // PyQIR currently requires named blocks
+  int blockCounter = 0;
+  for (llvm::Function &func : *llvmModule)
```

Comment:
Could use `llvm::enumerate` here if you don't care if the numbers for a sequence without any "holes".

Reply:
Do you mean something like this?
```suggestion
  for (llvm::Function &func : *llvmModule)
    for (auto &[blockIdx, block] : llvm::enumerate(func))
      if (!block.hasName())
        block.setName(std::to_string(blockIdx));
```

Reply:
After thinking about this, without deep diving into each backend's use of PyQIR, it would be difficult to know if this is OK, so I'm inclined to stick with the conservative path and leave it like it is, where each block gets a unique name.

Reply:
OK, I will add a comment.

Reply:
Added comment in 1f08eeee.